### ISSUE TITLE
Ephemeral nodes cannot access hosts on the network if running in a container

### ISSUE THREAD

**Issue by pyr0hu** [tailscale/tailscale#2970]
Hello!

I want to use ephemeral nodes for CI runners. I am using GitLab runner with autoscaling on AWS (therefore using docker). I have my own CI image which includes tailscale. When starting a runner, I register the node with `tailscale up --authkey $AUTHKEY --accept-routes`. Docker is using `NET_ADMIN` capabilities and binding the `/dev/net/tun` device.

First thing I noticed that I cannot access infra servers (e.g. staging) from these nodes. No response from ping. When I do a `tailscale down` and then `tailscale up` again, ping responds for ONE packet, all the other are lost. The IP resolution and dns resolution works properly, but I cannot reach the servers. 

If I use a general reusable authkey, everything works and the servers can be reached from the CI runner containers.

I found in the docs that ephemeral nodes only have IPv6 address, could that be the issue? I am using Alpine linux base images and those don't have IPv6 enabled.

**Comment by DentonGentry** [comment]
Tailscale is an overlay network, which means that when you access a Tailscale IP address like `fd7a:115c:a1e0:ab12:4843:cd96:3:2` or `100.99.98.97` that your data goes into an IP packet which is immediately encrypted and prepended with a Wireguard header. The encrypted blob is then placed inside another, entirely separate outer IP header.

The inner IP header inside the binary blob can be IPv6 which then gets carried through an outer IPv4 network.

My guess would be that something in the actions the CI is running is trying to contact an IPv4 Tailscale address like 100.99.98.97. That won't work, as an ephemeral node only gets an IPv6 address. You can use `tailscale ip -6 hostname` to look up the IPv6 address of the thing being contacted.

The symptom which doesn't fit is having one successful ping response and then failure. Having even one ping response must mean that the IPv6 Tailscale address is being pinged.

**Comment by bradfitz** [comment]
You'll only be able to reach peers over their IPv6 addresses.

7fe6ecf165bdb1525151bcbc8b6f5f92e05e0670 for #2268 gave peers MagicDNS names if they _only_ had IPv6 addresses (so e.g. others nodes could reach ephemeral nodes by name), but perhaps we never did the reverse: if you're an IPv6-only node yourself, we either shouldn't tell you about IPv4 addresses of your peers, or we should at least fix MagicDNS to return AAAA records for your peers.

/cc @danderson

**Comment by pyr0hu** [comment]
If I try to ping6 <ipv6>, I get these errors: `ping6: sendto: Network unreachable` . I'm starting to think this must be a docker ipv6 issue.

Do I have to set up anything in the containers to enable routing to ipv6? 

I've seen the docs creating Dockerfiles for AWS lambda, GitHub runners and all are using Alpine linux, so this shouldn't be an issue with the base image. Therefore my best guess would be something to do with Docker configuration.

**Comment by pyr0hu** [comment]
Update: Found an interesting warning in tailscaled logs that ip6tables command was not found. On a local container, installed the package and I could reach a host through ipv6! Gonna try out that on the CI.

**Comment by pyr0hu** [comment]
Success! Had to add to docker parameter: `--sysctl net.ipv6.conf.all.disable_ipv6=0` and added a 1sec sleep after connecting to Tailscale.

TLDR: It was a docker/container misconfiguration , now the hosts can be reached with their IPv6 addresses.
